### PR TITLE
refactor: Share sql lab query

### PR DESCRIPTION
**Pull request by AAfghahi**
### SUMMARY
Changed the share SQL Lab Query button into a TypeScript file using react hooks. 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
When Query is not saved:
Screen Shot 2021-02-25 at 5 45 38 PM
When Query is saved: 
Screen Shot 2021-02-25 at 5 45 49 PM
### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by AAfghahi** [comment]
Changed from: 

https://github.com/apache/superset/pull/13349

File: superset-frontend/spec/javascripts/sqllab/ShareSqlLabQuery_spec.jsx
Comment (eschutho) [inline superset-frontend/spec/javascripts/sqllab/ShareSqlLabQuery_spec.jsx:132]: if you want to dry this up, you can move the provider into a wrapper: https://testing-library.com/docs/react-testing-library/setup#custom-render

File: superset-frontend/src/SqlLab/components/ShareSqlLabQuery.tsx
Comment (eschutho) [inline superset-frontend/src/SqlLab/components/ShareSqlLabQuery.tsx:80]: do you know if this would be null or undefined? Whichever it is, let's add it to the interface.

File: superset-frontend/src/SqlLab/components/ShareSqlLabQuery.tsx
Comment (eschutho) [inline superset-frontend/src/SqlLab/components/ShareSqlLabQuery.tsx:136]: this was existing code, but since we're just passing the arg right on through, we should be able to just pass the function to the prop, like `getText={getCopyUrl}`

**Review by eschutho — COMMENTED** [review]
looks good. just a few small suggestions.

File: superset-frontend/spec/javascripts/sqllab/ShareSqlLabQuery_spec.jsx
Comment (AAfghahi) [reply superset-frontend/spec/javascripts/sqllab/ShareSqlLabQuery_spec.jsx:132]: Thank you, this looks a lot more elegant.

File: superset-frontend/src/SqlLab/components/ShareSqlLabQuery.tsx
Comment (AAfghahi) [reply superset-frontend/src/SqlLab/components/ShareSqlLabQuery.tsx:80]: just to clarify, you mean the remoteId

Comment (AAfghahi) [reply superset-frontend/src/SqlLab/components/ShareSqlLabQuery.tsx:80]: Looking through SqlEditor, it seems like database and latestQuery have a defaultProp of null. Which makes me think that this would be null type as well.

Comment (AAfghahi) [reply superset-frontend/src/SqlLab/components/ShareSqlLabQuery.tsx:80]: Ok, I changed it to null and tested it and the intended functionality is there.

**Comment by AAfghahi** [comment]
> looks good. just a few small suggestions.

Changes made! Thank you for the edits!

**Comment by junlincc** [comment]
@AAfghahi tooltip text "Save the query to copy the link" seems unclear to me. 😔

**Comment by Steejay** [comment]
@junlincc thanks for the feedback. Can you tell me more about why this copy is unclear to you? The action to copy is blocked until the user saves the query.

**Comment by junlincc** [comment]
My impression from the tooltip text is, this button is for user to save the query to 'something'. 
I didn't get the idea that there's a clear order of save the query first, then the button will be activated, then I can copy to link. 
I think what's missing is the button activation/enabling part.

**Comment by Steejay** [comment]
I see. Though if the confusion is "Save the query to (location)" I dont think the "copy the link" portion sounds like a location since there's nowhere in the product called "copy the link". The copy in this tool tip is stating the action needed (first) for the desired function to happen (second). 

We could alter it to read "Save the query to enable this feature" if that's more clear.

**Comment by junlincc** [comment]
> "Save the query to enable this feature"

seems a bit long, but better. "Save the query to enable"? but your call. 🙏 thanks

**Comment by kgabryje** [comment]
How about "Saving the query required"?

**Comment by Steejay** [comment]
@junlincc yes noted. guidelines are in progress for various undefined patterns that currently exist in Superset.

**Comment by Steejay** [comment]
@kgabryje thanks for the suggestion! Micro copy should try to use an active voice to communicate directions clearly. In this case we want to state the _action_ needed and the _outcome_ of the action to the user. 


@junlincc 
re "Save the query to enable this feature"

"enable" is a transitive verb (requires an object) so including the object ("feature") would make this copy most clear. 

we could wrap the text to avoid a long tooltip. otherwise im not concerned about the length of this tooltip. 
Screen Shot 2021-03-17 at 9 33 18 AM

**Comment by AAfghahi** [comment]
This is what it looks like unwrapped: 

Screen Shot 2021-03-17 at 2 12 04 PM

**Comment by Steejay** [comment]
Wrapped looks good to me.

**Review by hughhhh — APPROVED** [review]

**Pull request merged**
